graphblas/io.py

```diff
@@ -624,8 +666,30 @@ def mmwrite(target, matrix, *, comment="", field=None, precision=None, symmetry=
         from scipy.io import mmwrite
     except ImportError:  # pragma: no cover (import)
         raise ImportError("scipy is required to write Matrix Market files") from None
+    engine = engine.lower()
+    if engine in {"auto", "fmm", "fast_matrix_market"}:
+        try:
+            from fast_matrix_market import mmwrite  # noqa: F811
+        except ImportError:  # pragma: no cover (import)
+            if engine != "auto":
+                raise ImportError(
+                    "fast_matrix_market is required to write Matrix Market files "
+                    f'using the "{engine}" engine'
```

Comment:
Not sure to what extent DRY principle is useful here, but perhaps there is scope for generalising this check for suitable import as some utility function (as a separate PR).

Reply:
I'm open to such a PR :)

You may notice that my style sometimes doesn't adhere to DRY for small, straightforward (to me) things, but I suspect there are cases where things could be improved.